diff --git a/src/3rd_party/faiss/VectorTransform.cpp b/src/3rd_party/faiss/VectorTransform.cpp
index 103b091..718c162 100644
--- a/src/3rd_party/faiss/VectorTransform.cpp
+++ b/src/3rd_party/faiss/VectorTransform.cpp
@@ -132,7 +132,7 @@ const float *fvecs_maybe_subsample(
   return x_subset;
 }
 
-#if 1 // def __SSE__
+#ifdef __SSE__
 // reads 0 <= d < 4 floats as __m128
 static inline __m128 masked_read(int d, const float *x)
 {
@@ -175,10 +175,14 @@ float fvec_norm_L2sqr(const float *  x,
   return  _mm_cvtss_f32(msum1);
 }
 #else
-// scalar implementation
+// portable scalar implementation
 float fvec_norm_L2sqr(const float *x, size_t d)
 {
-  return fvec_norm_L2sqr_ref(x, d);
+  float sum = 0;
+  for (size_t i = 0; i < d; i++) {
+    sum += x[i] * x[i];
+  }
+  return sum;
 }
 #endif
 
